users/app.py

```diff
@@ -37,20 +41,60 @@ def get_authorization():
     return jsonify(edit_access=authorized)
 
 
-@app.route('/v1/', methods=['PUT'])
-def add_update_user():
-    """Add or update the user in the db and returns new user object."""
-    user = request.getJSON()
-    if user is None:
-        # TODO(mukobi) validate the user object has everything it needs
-        return jsonify(error="You must supply a valid user in the body")
-    # TODO(mukobi) add or update the user in the database
-    added_new_user = True
-    # TODO(mukobi) get the new user object from the db to return
-    user_object = {
-        "user_id": "0", "username": "Dummy User", "edit_access": False
-    }
-    return jsonify(user_object), (201 if added_new_user else 200)
+@app.route('/v1/authenticate', methods=['POST'])
+def authenticate_user():
```

Comment:
maybe call this get_and_authenticate_user or authenticate_and_get_user

Reply:
Done https://github.com/knative-portability/large-events/pull/31/commits/f2a8ccb636eee51d1e0c0a367c29b136ddf9486e